ansible-ipi-install/roles/installer/tasks/57_check_vip_status.yml

```diff
@@ -0,0 +1,20 @@
+---
+- name: Capture API VIP status
+  uri:
+    url: "{{ item }}" 
+    validate_certs: no
+  register: return_status
+  ignore_errors: yes
+  loop:
+    - https://{{ apivip }}:6443/readyz
+    - https://{{ ingressvip}}:433/readyz
+  when: ('Expected output is failed with a status code of -1 as VIP's should not be active' | warn_me()) 
+  tags: vip_status
+
+- name: Fail if Ingress VIP already exists
+  fail:
+    msg: |
+      The API VIP or Ingress VIP is still active in your cluster
```

Comment:
I prefer to explicitly warn the user which is the problematic VIP. Otherwise, the user will need to perform the steps manually to check which one is the on that is still up.

Reply:
@e-minguez When you run this code, Ansible gives you output in the "Verify if API or Ingress VIP is running" task of which is the bad one. For example:

Output when API or VIP are online (giving values for both online here)
```
TASK [Verify if API or Ingress VIP is running.] *********************************************************************************************************
Tuesday 02 June 2020  07:45:14 -0500 (0:00:07.941)       0:00:07.963 ********** 
[WARNING]: Expected output is failed with a status code of -1 as VIPs should not be active

ok: [worker-0.kni4.cloud.lab.eng.bos.redhat.com] => (item=https://api.kni4.cloud.lab.eng.bos.redhat.com:6443/readyz)
failed: [worker-0.kni4.cloud.lab.eng.bos.redhat.com] (item=https://foo.apps.kni4.cloud.lab.eng.bos.redhat.com:443/readyz) => {"ansible_loop_var": "item", "cache_control": "private, max-age=0, no-cache, no-store", "changed": false, "content": "<html>\n  <head>\n    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n\n  <style type=\"text/css\">\n  /*!\n   * Bootstrap v3.3.5 (http://getbootstrap.com)\n   * Copyright 2011-2015 Twitter, Inc.\n   * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)\n   */\n  /*! normalize.css v3.0.3 | MIT License | github.com/necolas/normalize.css */\n  html {\n    font-family: sans-serif;\n    -ms-text-size-adjust: 100%;\n    -webkit-text-size-adjust: 100%;\n  }\n  body {\n    margin: 0;\n  }\n  h1 {\n    font-size: 1.7em;\n    font-weight: 400;\n    line-height: 1.3;\n    margin: 0.68em 0;\n  }\n  * {\n    -webkit-box-sizing: border-box;\n    -moz-box-sizing: border-box;\n    box-sizing: border-box;\n  }\n  *:before,\n  *:after {\n    -webkit-box-sizing: border-box;\n    -moz-box-sizing: border-box;\n    box-sizing: border-box;\n  }\n  html {\n    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);\n  }\n  body {\n    font-family: \"Helvetica Neue\", Helvetica, Arial, sans-serif;\n    line-height: 1.66666667;\n    font-size: 13px;\n    color: #333333;\n    background-color: #ffffff;\n    margin: 2em 1em;\n  }\n  p {\n    margin: 0 0 10px;\n    font-size: 13px;\n  }\n  .alert.alert-info {\n    padding: 15px;\n    margin-bottom: 20px;\n    border: 1px solid transparent;\n    background-color: #f5f5f5;\n    border-color: #8b8d8f;\n    color: #363636;\n    margin-top: 30px;\n  }\n  .alert p {\n    padding-left: 35px;\n  }\n  a {\n    color: #0088ce;\n  }\n\n  ul {\n    position: relative;\n    padding-left: 51px;\n  }\n  p.info {\n    position: relative;\n    font-size: 15px;\n    margin-bottom: 10px;\n  }\n  p.info:before, p.info:after {\n    content: \"\";\n    position: absolute;\n    top: 9%;\n    left: 0;\n  }\n  p.info:before {\n    content: \"i\";\n    left: 3px;\n    width: 20px;\n    height: 20px;\n    font-family: serif;\n    font-size: 15px;\n    font-weight: bold;\n    line-height: 21px;\n    text-align: center;\n    color: #fff;\n    background: #4d5258;\n    border-radius: 16px;\n  }\n\n  @media (min-width: 768px) {\n    body {\n      margin: 4em 3em;\n    }\n    h1 {\n      font-size: 2.15em;}\n  }\n\n  </style>\n  </head>\n  <body>\n    <div>\n      <h1>Application is not available</h1>\n      <p>The application is currently not serving requests at this endpoint. It may not have been started or is still starting.</p>\n\n      <div class=\"alert alert-info\">\n        <p class=\"info\">\n          Possible reasons you are seeing this page:\n        </p>\n        <ul>\n          <li>\n            <strong>The host doesn't exist.</strong>\n            Make sure the hostname was typed correctly and that a route matching this hostname exists.\n          </li>\n          <li>\n            <strong>The host exists, but doesn't have a matching path.</strong>\n            Check if the URL path was typed correctly and that the route was created using the desired path.\n          </li>\n          <li>\n            <strong>Route and path matches, but all pods are down.</strong>\n            Make sure that the resources exposed by this route (pods, services, deployment configs, etc) have at least one pod running.\n          </li>\n        </ul>\n      </div>\n    </div>\n  </body>\n</html>\n", "content_type": "text/html", "elapsed": 0, "item": "https://foo.apps.kni4.cloud.lab.eng.bos.redhat.com:443/readyz", "msg": "Status code was 503 and not [200]: HTTP Error 503: Service Unavailable", "pragma": "no-cache", "redirected": false, "status": 503, "url": "https://foo.apps.kni4.cloud.lab.eng.bos.redhat.com:443/readyz"}
...ignoring

TASK [Fail if API VIP or Ingress VIP already exists] ****************************************************************************************************
Tuesday 02 June 2020  07:45:20 -0500 (0:00:06.287)       0:00:14.251 ********** 
fatal: [worker-0.kni4.cloud.lab.eng.bos.redhat.com]: FAILED! => {"changed": false, "msg": "The API VIP or Ingress VIP is already running, please ensure you don't have\nany rogue servers up and running.\n"}

```

Output when API or VIP are not online

```
TASK [Verify if API or Ingress VIP is running.] *********************************************************************************************************
Tuesday 02 June 2020  07:43:54 -0500 (0:00:05.825)       0:00:05.849 ********** 
[WARNING]: Expected output is failed with a status code of -1 as VIPs should not be active

failed: [worker-0.kni4.cloud.lab.eng.bos.redhat.com] (item=https://api.kni44.cloud.lab.eng.bos.redhat.com:6443/readyz) => {"ansible_loop_var": "item", "changed": false, "content": "", "elapsed": 0, "item": "https://api.kni44.cloud.lab.eng.bos.redhat.com:6443/readyz", "msg": "Status code was -1 and not [200]: Request failed: <urlopen error [Errno -2] Name or service not known>", "redirected": false, "status": -1, "url": "https://api.kni44.cloud.lab.eng.bos.redhat.com:6443/readyz"}
failed: [worker-0.kni4.cloud.lab.eng.bos.redhat.com] (item=https://foo.apps.kni44.cloud.lab.eng.bos.redhat.com:443/readyz) => {"ansible_loop_var": "item", "changed": false, "content": "", "elapsed": 0, "item": "https://foo.apps.kni44.cloud.lab.eng.bos.redhat.com:443/readyz", "msg": "Status code was -1 and not [200]: Request failed: <urlopen error [Errno -2] Name or service not known>", "redirected": false, "status": -1, "url": "https://foo.apps.kni44.cloud.lab.eng.bos.redhat.com:443/readyz"}
...ignoring

TASK [Fail if API VIP or Ingress VIP already exists] ****************************************************************************************************
Tuesday 02 June 2020  07:44:00 -0500 (0:00:06.718)       0:00:12.567 ********** 
skipping: [worker-0.kni4.cloud.lab.eng.bos.redhat.com]

```

Reply:
It is still confusing to me because I don't know exactly which VIP is failing on the first example.

Reply:
@e-minguez This is why I added that WARNING because it is confusing. In fact the uri module expects error when it sees a value of `-1` and we see success when we see `-1`. This is the difficulty of using `-1` as its counter intuitive to what you'd expect.